File: StPrintQueue.Db/Entities/Job.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace StPrintQueue.Db.Entities
{
    public class Job
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Duration { get; set; }
        public DateTime? StartTime { get; set; }
        public JobStatus Status { get; set; }
    }
}

File: StPrintQueue/src/app/models/Job.ts
import { JobStatus } from './JobStatus';
import * as moment from 'moment';

export class Job
{
    id: number;
    name: string;
    duration: number;
    status: JobStatus;

    startTime: Date;
    endTime: Date;
    constructor() {

    }

    static getRemainingSeconds(job: Job) : number {
        //startTime + duration(s) = estimated endTime
        //endTime - currentTime = actual timestamp in milliseconds. /1000 to get it in seconds.
        if(job.startTime) {
            let actualSecondsLeft  = moment(job.startTime).add(job.duration, 's').diff(moment()) /1000;
            return actualSecondsLeft;
        }
    }
}